wdl-analysis/src/handlers/completions.rs

```diff
@@ -0,0 +1,453 @@
+                | SyntaxKind::ConditionalStatementNode => {
+                    add_keyword_completions(&WORKFLOW_ITEM_EXPECTED_SET, &mut items);
+                    if let Some(scope) = document.find_scope_by_position(offset.into()) {
+                        add_scope_completions(scope, &mut items);
+                    }
+                    add_stdlib_completions(&mut items);
+                    add_struct_completions(document, &mut items);
+                    add_namespace_completions(document, &mut items);
+                    add_callable_completions(document, &mut items);
+                    break;
+                }
+
+                SyntaxKind::TaskDefinitionNode => {
+                    add_keyword_completions(&TASK_ITEM_EXPECTED_SET, &mut items);
+                    if let Some(scope) = document.find_scope_by_position(offset.into()) {
+                        add_scope_completions(scope, &mut items);
+                    }
+                    add_stdlib_completions(&mut items);
+                    break;
+                }
+                SyntaxKind::RootNode => {
+                    add_keyword_completions(
+                        &TOP_EXPECTED_SET.union(TokenSet::new(&[Token::VersionKeyword as u8])),
+                        &mut items,
+                    );
+                    add_struct_completions(document, &mut items);
+                    add_namespace_completions(document, &mut items);
+                    break;
+                }
+                _ => current = node.parent(),
+            }
+        }
+    }
+
+    if partial_word.is_empty() {
+        return Ok(items);
+    }
+
+    let items = items
+        .into_iter()
+        .filter(|item| item.label.starts_with(&partial_word))
+        .collect();
+
+    Ok(items)
+}
+
+/// Generates completion items for WDL keywords based on the provided token set.
+///
+/// Converts raw token values to completion items with appropriate labels,
+/// kinds, and descriptions.
+fn add_keyword_completions(token_set: &TokenSet, items: &mut Vec<CompletionItem>) {
+    items.extend(token_set.iter().map(|raw| {
+        let token = Token::from_raw(raw);
+        let label = token
+            .describe()
+            .trim_start_matches("`")
+            .split("`")
+            .next()
+            .unwrap_or("")
+            .to_string();
+
+        CompletionItem {
+            label: label.to_string(),
+            kind: Some(CompletionItemKind::KEYWORD),
+            detail: Some(token.describe().to_string()),
+            ..Default::default()
+        }
+    }))
+}
+
+/// Adds completions for member access expressions.
+///
+/// Handles different types of member access completions:
+/// - Namespace access
+/// - Struct member access
+/// - Call output access
+/// - Pair element access (when accessing `.left` and `.right` of pair types)
+///
+/// For namespace access, it directly looks up the identifier before the dot.
```

Comment:
Can you give a little more on what `node` is here? What's expected to be contained within?